core/src/main/java/io/atomix/core/impl/ClasspathScanningAtomixRegistry.java

```diff
@@ -43,25 +44,26 @@ public ClasspathScanningAtomixRegistry(ClassLoader classLoader, Class<? extends
 
   public ClasspathScanningAtomixRegistry(ClassLoader classLoader, Collection<Class<? extends NamedType>> types) {
     final String scanSpec = System.getProperty("io.atomix.scanSpec");
-    final FastClasspathScanner classpathScanner = scanSpec != null ?
-        new FastClasspathScanner(scanSpec).addClassLoader(classLoader) :
-        new FastClasspathScanner().addClassLoader(classLoader);
+    final ClassGraph classGraph = scanSpec != null ?
+            new ClassGraph().enableClassInfo().whitelistPackages(scanSpec).addClassLoader(classLoader) :
+            new ClassGraph().enableClassInfo().addClassLoader(classLoader);
 
+    final ScanResult scanResult = classGraph.scan();
     for (Class<? extends NamedType> type : types) {
       Map<String, NamedType> registrations = new ConcurrentHashMap<>();
-      classpathScanner.matchClassesImplementing(type, clazz -> {
-        if (!Modifier.isAbstract(clazz.getModifiers()) && !Modifier.isPrivate(clazz.getModifiers())) {
-          NamedType instance = newInstance(clazz);
-          NamedType oldInstance = registrations.put(instance.name(), instance);
-          if (oldInstance != null) {
-            LOGGER.warn("Found multiple types with name={}, classes=[{}, {}]", instance.name(),
-                oldInstance.getClass().getName(), instance.getClass().getName());
-          }
+      scanResult.getClassesImplementing(type.getName()).forEach(classInfo -> {
+        if (classInfo.isInterface() || classInfo.isAbstract() || Modifier.isPrivate(classInfo.getModifiers())) {
```

Comment:
Because you called `.enableClassInfo()` above, but not `.ignoreClassVisibility()`, you will not get non-public classes here anyway.

Reply:
good to know but I'll leave it just incase someone checks the Atomix source, easier to see what's expected.